include/wifi_hal_extender.h

```diff
@@ -101,627 +106,1377 @@ typedef struct _wifi_associated_dev_rate_info_rx_stats {
+ */
+INT wifi_steering_clientRemove(UINT steeringgroupIndex, INT apIndex, mac_address_t client_mac);
+
+/**
+ * @brief Initiates an instant client RSSI measurement.
+ *
+ * This function initiates an instant client RSSI measurement. The recommended
+ * method of performing this measurement is to send five NULL Wi-Fi frames to
+ * the client and average the RSSI of the ACK frames returned. This averaged
+ * RSSI value should be sent back using the `WIFI_STEERING_EVENT_RSSI`
+ * steering event type.
+ *
+ * Instant measurement improves user experience by not reacting to
+ * false-positive RSSI crossings.
+ *
+ * @param[in] steeringgroupIndex Steering group index.
+ * @param[in] apIndex            Access Point index.
+ * @param[in] client_mac         Client MAC address.
+ *
+ * @returns The status of the operation.
+ * @retval RETURN_OK  If successful.
+ * @retval RETURN_ERR If any error is detected.
+ */
+INT wifi_steering_clientMeasure(UINT steeringgroupIndex, INT apIndex, mac_address_t client_mac);
+
+/**
+ * @brief Initiates a client disconnect.
+ *
+ * @param[in] steeringgroupIndex Steering group index.
+ * @param[in] apIndex            Access Point index.
+ * @param[in] client_mac         Client MAC address.
+ * @param[in] type               Disconnect type.
+ * @param[in] reason             Reason code to provide in the deauthentication
+ *                               or disassociation frame.
+ *
+ * @returns The status of the operation.
+ * @retval RETURN_OK  If successful.
+ * @retval RETURN_ERR If any error is detected.
+ *
+ * @see https://supportforums.cisco.com/document/141136/80211-association-status-80211-deauth-reason-codes
+ */
+INT wifi_steering_clientDisconnect(UINT steeringgroupIndex, INT apIndex, mac_address_t client_mac, wifi_disconnectType_t type, UINT reason);
+
+/** @} */  //END OF GROUP WIFI_HAL_APIS
+
+/**
+ * @addtogroup WIFI_HAL_TYPES
+ * @{
+ 
+#ifdef WIFI_HAL_VERSION_3_PHASE2
+/**
+ * @brief Callback function invoked when a STA sends a BTM query.
+ *
+ * This callback function is invoked when a station (STA) sends a BSS
+ * Transition Management (BTM) query message to a Virtual Access Point (VAP)
+ * in the gateway. The driver will use the frame returned from this function
+ * to process the response to the query.
+ *
+ * @param[in] apIndex        Access Point index.
+ * @param[in] peerMac        MAC address of the peer STA.
+ * @param[in] query          Pointer to the received BTM query frame.
+ * @param[in] inMemSize      Size of the memory allocated for the response frame.
+ * @param[out] request       Pointer to a `wifi_BTMRequest_t` structure to store
+ *                          the BTM request frame.
+ *
+ * @returns The status of the operation.
+ * @retval WIFI_HAL_SUCCESS If successful.
+ * @retval WIFI_HAL_ERROR   If any error is detected.
+ */
+typedef INT (*wifi_BTMQueryRequest_callback)(UINT apIndex, mac_address_t peerMac, wifi_BTMQuery_t *query, UINT inMemSize, wifi_BTMRequest_t *request);
```

Comment:
Declaration of callback function wifi_BTMResponse_callback is removed, needs to be added here.
/**

@brief This call back is invoked when a STA responds to a BTM Request
from the gateway.
@param apIndex - Access Point Index.
@param peerMACAddress - MAC address of the peer the response was received
from.
@param in_struct - Response frame received from a non-AP STA.
@return The status of the operation.
@RetVal RETURN_OK if successful.
@RetVal RETURN_ERR if any error is detected.
@execution Synchronous
https://github.com/sideeffect None
@note This function must not suspend and must not invoke any blocking system
calls.
/
typedef INT ( wifi_BTMResponse_callback)(UINT apIndex,
mac_address_t peerMac,
wifi_BTMResponse_t *response);